numba_dpex/tests/experimental/kernel_iface/spv_overloads/test_atomic_fetch_phi.py

```diff
@@ -14,32 +16,297 @@
     no_bool=True, no_float16=True, no_none=True, no_complex=True
 )
 
+list_of_int_dtypes = get_all_dtypes(
+    no_bool=True, no_float=True, no_none=True, no_complex=True
+)
+
+list_of_float_dtypes = get_all_dtypes(
+    no_bool=True, no_int=True, no_float16=True, no_none=True, no_complex=True
+)
+
 
 @pytest.fixture(params=list_of_supported_dtypes)
-def input_arrays(request):
+def input_arrays_for_add(request):
     # The size of input and out arrays to be used
     N = 10
     a = dpnp.ones(N, dtype=request.param)
     b = dpnp.zeros(N, dtype=request.param)
     return a, b
 
 
+@pytest.fixture(params=list_of_supported_dtypes)
```

Comment:
We do not need to create separate fixtures for each fetch_phi function. Also all the tests to do positive validation can be combined.

How about something like the following snippet?

```python

list_of_fetch_phi_funcs = [
    "fetch_add",
    "fetch_sub",
]

@pytest.fixture(params=list_of_fetch_phi_funcs)
def fetch_pi_fn(request):
    return request.param

@pytest.fixture(params=list_of_supported_dtypes)
def input_arrays(request):
    # The size of input and out arrays to be used
    N = 10
    a = dpnp.arange(N, dtype=request.param)
    b_np = np.ones(N) * N + 1
    b = dpnp.asarray(b_np, dtype=request.param)
    return a, b

@pytest.mark.parametrize("ref_index", [0, 5])
def test_fetch_phi_fn(input_arrays, ref_index, fetch_pi_fn):
    @dpex_exp.kernel
    def _kernel(a, b, ref_index):
        i = dpex.get_global_id(0)
        v = AtomicRef(b, index=ref_index)
        getattr(v, fetch_pi_fn)(a[i])

    a, b = input_arrays_for_sub
    dpex_exp.call_kernel(_kernel, dpex.Range(10), a, b, ref_index)

    # Verify that `a` accumulated at b[ref_index] by kernel
    # matches the `a` accumulated at  b[ref_index+1] using Python
    for i in range(a.size):
        v = AtomicRef(b, index=ref_index+1)
        getattr(v, fetch_pi_fn)(a[i])

    assert b[ref_index] == b[ref_index+1]


```

Reply:
Reduced tests to one positive and one negative for all fetch_phi's.